pkg/cortex/modules.go

```diff
@@ -193,7 +194,24 @@ func (t *Cortex) initQuerier(cfg *Config) (err error) {
 		return
 	}
 
-	queryable, engine := querier.New(cfg.Querier, t.distributor, t.store)
+	var store querier.ChunkStore
+	if cfg.Ingester.V2.Enabled {
+		s3cfg := s3.Config{
+			Bucket:    cfg.Ingester.V2.S3Bucket,
+			Endpoint:  cfg.Ingester.V2.S3Endpoint,
+			AccessKey: cfg.Ingester.V2.S3Key,
+			SecretKey: cfg.Ingester.V2.S3Secret,
+			Insecure:  cfg.Ingester.V2.S3Insecure,
+		}
+		store, err = querier.NewBlockQuerier(s3cfg, cfg.Ingester.V2.TSDBDir, prometheus.DefaultRegisterer)
```

Comment:
Reusing the same `TSDBDir`, is there the change to hit into race conditions when cortex runs in single binary mode (thus both ingester and querier run on the same node)?

_Not sure, I'm just asking your perspective._

Reply:
I've just tested it and looks not working. Have you tried it?

If the ingester and querier TSDB dir is the same, the querier fails with:
`failed to find/create user db`

And the root error is:
`unexpected corrupted block:map[01DQW26PST142CN7KK8HHZ4ZHX:open /tmp/tsdb/fake/01DQW26PST142CN7KK8HHZ4ZHX/chunks: no such file or directory 01DQW36ZHCBTP4V1K96RWWJRQE:open /tmp/tsdb/fake/01DQW36ZHCBTP4V1K96RWWJRQE/chunks: no such file or directory]`

If I change the base dir for the querier to another dir, it works.

Reply:
Fixed